webpack/scenes/ContentViews/Details/Versions/BulkDelete/Steps/ConfirmBulkDelete.js

```diff
@@ -0,0 +1,103 @@
+
+export default () => {
+  const {
+    versions, selectedCVForHosts, selectedEnvForHosts, selectedCVForAK, selectedEnvForAK,
+  } = useContext(BulkDeleteContext);
+  const { results: contentViewResults = [] } = useSelector(selectContentViews);
+  const versionList = getVersionListString(versions);
+  const numberOfActivationKeys = getNumberOfActivationKeys(versions);
+  const numberOfHosts = getNumberOfHosts(versions);
+  const affectedVersions = versions.filter(({ environments }) => !!environments.length);
+  const affectedVersionsListString = getVersionListString(affectedVersions);
+  const environments = getEnvironmentList(versions);
+
+  const pluralEnvironments = getNumberOfEnviroments(versions) > 1;
+  return (
+    <>
+      <WizardHeader
+        title={__('Review Details')}
+      />
+      <ActionSummary
+        title={__('Versions')}
+        text={<FormattedMessage
+          id="bulk-delete-summary-versions"
+          values={{
+            versionList,
+            versionOrVersions: versions.length > 1 ?
+              __('Versions') : __('Version'),
+          }}
+          defaultMessage={__('{versionOrVersions} {versionList} will be deleted and will no longer be available for promotion.')}
+        />}
+      />
+      {!!affectedVersions.length &&
+        <div>
+          <ActionSummary
+            title={__('Environments')}
+            text={
+              <FormattedMessage
+                id="bulk-delete-summary-environments"
+                values={{ versionList: affectedVersionsListString, versionOrVersions: affectedVersions.length > 1 ? __('Versions ') : __('Version ') }}
+                defaultMessage={pluralEnvironments ?
+                  __('{versionOrVersions} {versionList} will be removed from the following environments:') :
+                  __('{versionOrVersions} {versionList} will be removed from the following environment:')}
+              />
+            }
+          />
+          <Flex>
+            {environments.map(({ name, id }) => (
+              <FlexItem key={id}>
+                <Label isTruncated color="purple" href={`/lifecycle_environments/${id}`}>{name}</Label>
+              </FlexItem>))
+            }
+          </Flex>
+        </div>
+      }
+      {!!numberOfActivationKeys &&
+        <ActionSummary
+          title={__('Activation Keys')}
+          text={
+            `${numberOfActivationKeys} ${numberOfActivationKeys > 1 ?
+              __('activation keys will be assigned to content view') :
+              __('activation key will be assigned to content view')} '${contentViewResults
+                .find(({ id }) => id === selectedCVForAK)?.name || ''}' ${__('in')}`
+          }
+          selectedEnv={environments
+            .find(({ id }) => id === first(selectedEnvForAK).id)}
+        />}
+      {!!numberOfHosts &&
+        <ActionSummary
+          title={__('Hosts')}
+          text={
+            `${numberOfHosts} ${numberOfHosts > 1 ?
```

Comment:
```suggestion
            `${numberOfHosts} ${numberOfHosts !== 1 ?
```
Technically the definition of plural should be `!== 1` or else you might get things like `0 host`.

Reply:
I get you, but the number will never be 0 as directly above I have  ` {!!numberOfHosts &&` 
So the presumption is that if it's not 0, it will either be 1 or more. 
Now you could argue that NaN could potentially get through, but fortunately that value doesn't translate through the REST api, so it is very unlikely to be seen (unless Redux does something weird).

Reply:
True for this case. However, I did notice that the logic to determine if words are plural is duplicated in a few places here (Which isn't necessarily bad!) But if you were ever going to extract that into a reusable function, you'd want to cover all edge cases.  I guess what I'm saying is I agree it's fine the way it is, but if you decide to reuse this logic that's something you'll have to account for.

Reply:
Agree.